pkg/api/validation/validation.go

```diff
@@ -273,8 +278,12 @@ func ValidateObjectMeta(meta *api.ObjectMeta, requiresNamespace bool, nameFn Val
 func ValidateObjectMetaUpdate(new, old *api.ObjectMeta) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 
+	if !RepairMalformedUpdates && new.UID != old.UID {
+		allErrs = append(allErrs, errs.NewFieldInvalid("uid", new.UID, "field is immutable"))
+	}
 	// in the event it is left empty, set it, to allow clients more flexibility
-	if len(new.UID) == 0 {
+	// TODO: remove this logic when we retire the clients that modify UID.
```

Comment:
sorry, one more request, add:
// Please do not copy this pattern elsewhere; validation functions should *not* be modifying the objects they are passed!